src/nodecache.h

```diff
@@ -13,55 +13,50 @@
 	every time a new tree is generated.
 */
 
-typedef std::pair<long, long> IndexPair;
-
 class NodeCache
 {
 public:
     NodeCache() { }
-    virtual ~NodeCache() { cache.clear(); }
+    virtual ~NodeCache() { }
 
     bool contains(const AbstractNode &node) const {
-        auto result = cache.find(node.index()); 
-        return result != cache.end() && result->second.second >= 0L;
+        auto result = this->cache.find(node.index()); 
+        return result != this->cache.end() && 
+            result->second.second >= 0L && 
+            this->rootString.size() >= result->second.second;
```

Comment:
What does that last check do? Will we ever generate indices that are beyond the size of the full string?

Reply:
Its mainly an added protection in case this method is called before NodeDumper finishes traversing, because the rootString value isn't set until the very end.
An empty string check would also do roughly the same thing.

Reply:
Good point